test/functional/lua/diagnostic_spec.lua

```diff
@@ -1918,5 +1918,31 @@ describe('vim.diagnostic', function()
         return {show_called, hide_called}
       ]])
     end)
+
+    it('triggers the autocommand when diagnostics are set', function()
+      eq(1, exec_lua [[
+        vim.g.diagnostic_autocmd_triggered = 0
+        vim.cmd('autocmd DiagnosticChanged * let g:diagnostic_autocmd_triggered = 1')
+        vim.api.nvim_buf_set_name(diagnostic_bufnr, "test | test")
+        vim.wait(100, function ()
```

Comment:
What's the rationale for `vim.wait()`? Seems to work fine without it.

Reply:
You're right! My bad, it seems it's unecessary. Fixed in 7ad910d.